core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -87,11 +86,19 @@ public ApplicationServer(T config) {
     this(config, createThreadPool(config));
   }
 
+  public ApplicationServer(T config, List<NamedURI> namedListeners) {
+    this(config, namedListeners, createThreadPool(config));
+  }
+
   public ApplicationServer(T config, ThreadPool threadPool) {
+    this(config, null, threadPool);
+  }
+
+  public ApplicationServer(T config, List<NamedURI> namedListeners, ThreadPool threadPool) {
```

Comment:
I find it strange that we parse non-named listeners inside `ApplicationServer` but have to pass named-listeners as an argument. Why can't we parse named-listeners inside `ApplicationServer` as well?

Reply:
this was the approach i took initially in #266 (see the main PR comment). we could do it that way.